src/chainlit/playground/providers/langchain.py

```diff
@@ -0,0 +1,88 @@
+from typing import List, Union
+
+from fastapi.responses import PlainTextResponse, StreamingResponse
+from langchain.schema.messages import (
+    AIMessage,
+    BaseMessageChunk,
+    HumanMessage,
+    SystemMessage,
+)
+
+from chainlit import input_widget
+from chainlit.playground.provider import BaseProvider
+from chainlit.prompt import PromptMessage
+from chainlit.sync import make_async
+
+
+class LangchainGenericProvider(BaseProvider):
+    from langchain.chat_models.base import BaseChatModel
+    from langchain.llms.base import LLM
+
+    llm: Union[LLM, BaseChatModel]
+
+    def __init__(
+        self,
+        id: str,
+        name: str,
+        llm: Union[LLM, BaseChatModel],
+        inputs: List[input_widget.InputWidget],
+        is_chat: bool = False,
+        is_automated: bool = False,
+    ):
+        super().__init__(
+            id=id,
+            name=name,
+            env_vars={},
+            inputs=inputs,
+            is_chat=is_chat,
+            is_automated=is_automated,
+        )
+        self.llm = llm
+
+    def prompt_message_to_langchain_message(self, message: PromptMessage):
+        content = "" if message.formatted is None else message.formatted
+        if message.role == "user":
+            return HumanMessage(content=content)
+        elif message.role == "assistant":
+            return AIMessage(content=content)
+        elif message.role == "system":
+            return SystemMessage(content=content)
```

Comment:
I think we should also handle FunctionMessage. Here is a small snippet to test it:

```py
from langchain import LLMMathChain
from langchain.agents import initialize_agent, Tool
from langchain.agents import AgentType
from langchain.chat_models import ChatOpenAI


llm = ChatOpenAI(temperature=0, model="gpt-3.5-turbo-0613")
llm_math_chain = LLMMathChain.from_llm(llm=llm, verbose=True)
tools = [
    Tool(
        name="Calculator",
        func=llm_math_chain.run,
        description="useful for when you need to answer questions about math"
    )
]

mrkl = initialize_agent(tools, llm, agent=AgentType.OPENAI_FUNCTIONS, verbose=True)
```

Reply:
done